### PR TITLE
Add detection api for Paddle 2.0

### PR DESCRIPTION
**Pull request by jerrywgz**
Migrate detection op(collect_fpn_proposals) from Paddle to PaddleDetection
Take collect_fpn_proposals as example

### CODE REVIEW

File: ppdet/modeling/ops/collect_fpn_proposals.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        type='collect_fpn_proposals',
+        inputs=inputs,
+        outputs=outputs,
+        attrs={'post_nms_topN': post_nms_top_n})
+    if rois_num_per_level is not None:
+        return output_rois, rois_num
+    return output_rois
+
+
+class CollectFpnProposals(layers.Layer):
```
Comment (willthefrog) [inline ppdet/modeling/ops/collect_fpn_proposals.py:131]: do not see the point of adding class interfaces, as none(?) of the OPs has associated weights...

Comment (jerrywgz) [reply ppdet/modeling/ops/collect_fpn_proposals.py]: Done, thx

File: ppdet/modeling/layers.py [hunk 2]
```diff
@@ -0,0 +1,482 @@
+import numpy as np
```
Comment (heavengate) [inline ppdet/modeling/layers.py:1]: add copyright

File: ppdet/modeling/ops.py [hunk 3]
```diff
@@ -1,482 +1,130 @@
...
 import numpy as np
-from numbers import Integral
-import paddle.fluid as fluid
-from paddle.fluid.dygraph.base import to_variable
-from ppdet.core.workspace import register, serializable
-from ppdet.py_op.target import generate_rpn_anchor_target, generate_proposal_target, generate_mask_target
-from ppdet.py_op.post_process import bbox_post_process
+from functools import reduce
 
+__all__ = ['collect_fpn_proposals', ]
```
Comment (heavengate) [inline ppdet/modeling/ops.py:27]: 这个要保留么，还是后面修改

File: ppdet/modeling/layers.py (hunk 2 above)
Comment (jerrywgz) [reply ppdet/modeling/layers.py:15]: Done, thx

File: ppdet/modeling/ops.py (hunk 3 above)
Comment (jerrywgz) [reply ppdet/modeling/ops.py]: 修改为了所有需要迁移的api名称

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**